src/parser/expression.rs

```diff
@@ -38,17 +38,24 @@ enum Precedence {
 impl From<&Token> for Precedence {
     fn from(token: &Token) -> Self {
         match token {
-            Token::Operator(Operator::Plus) | Token::Operator(Operator::Minus) => Precedence::Sum,
-            Token::Operator(Operator::Star) | Token::Operator(Operator::Slash) => {
-                Precedence::Product
-            }
+            Token::Operator(operator) => Self::from(operator),
             // TODO: Is this used?
             Token::LParenthesis => Precedence::Call,
             _ => Precedence::Lowest,
         }
     }
 }
 
+impl From<&Operator> for Precedence {
+    fn from(operator: &Operator) -> Self {
+        match operator {
+            Operator::Plus | Operator::Minus => Precedence::Sum,
```

Comment:
Does quil also support one or both of these as unary operators. e.g. `+2`, `-4`?

Reply:
Yes, `+` and `-` are both valid unary operators in quil. Are you getting at something with that question?

Reply:
I was wondering if there needed to be a separate case for `+`/`-` as unary operators, since they would need higher precedence than `*`/`/`. Though thinking about it now, I suppose the lexer could/should be handling that case, not the parser.

Reply:
Yeah, I think that makes more sense to handle with the lexer. Either way, tangential to the fix at hand, so I'll resolve.